services/slam/builtin/builtin.go

```diff
@@ -481,7 +481,35 @@ func (slamSvc *builtIn) GetPosition(ctx context.Context, name string) (spatialma
 	ctx, span := trace.StartSpan(ctx, "slam::builtIn::GetPosition")
 	defer span.End()
 
-	return nil, "", errors.New("unimplemented stub")
+	req := &pb.GetPositionNewRequest{Name: name}
+
+	resp, err := slamSvc.clientAlgo.GetPositionNew(ctx, req)
+	if err != nil {
+		return nil, "", errors.Wrap(err, "error getting SLAM position")
+	}
+	pose := spatialmath.NewPoseFromProtobuf(resp.GetPose())
+	componentReference := resp.GetComponentReference()
+	returnedExt := resp.Extra.AsMap()
+
+	// check if extra contains a quaternion. If it does, return the pose with the actual orientation
+	if val, ok := returnedExt["quat"]; ok {
+		q := val.(map[string]interface{})
+
+		valReal, ok1 := q["real"].(float64)
+		valIMag, ok2 := q["imag"].(float64)
+		valJMag, ok3 := q["jmag"].(float64)
+		valKMag, ok4 := q["kmag"].(float64)
+
+		if !ok1 || !ok2 || !ok3 || !ok4 {
+			slamSvc.logger.Debugf("quaternion given, but invalid format detected, %v, skipping quaternion transform", q)
```

Comment:
Is it actually acceptable for the slam algo GRPC server to not return the extra quat values expected?

Are we sure this shouldn't be an error now?

Reply:
i think you are right, since we assume builtin is only talking to one of our implemented algos this should throw an error

Reply:
Lets confirm in the slam channel

Reply:
Yes, this should be an error. Once this goes in the orbslam/cartographer modules, it will be clear this is expected to return the extra quat values.